WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentDisconnectedMessageView.swift

```diff
@@ -1,7 +1,8 @@
 import SwiftUI
 
 struct PointOfSaleCardPresentPaymentReaderDisconnectedMessageView: View {
-    let viewModel: PointOfSaleCardPresentPaymentReaderDisconnectedMessageViewModel
+    private let viewModel = PointOfSaleCardPresentPaymentReaderDisconnectedMessageViewModel()
+    @EnvironmentObject private var posModel: PointOfSaleAggregateModel
```

Comment:
One more small comment on this. I've seen recommendations that injecting large environment objects is recommended for larger views while keeping smaller views more independent. The `posModel.connectCardReader` could still be passed/called from the `TotalsView` rather than here.

Reply:
Done here 46b2e09 – do you know why this is recommended, or have any links? 

One reason might just be to make the leaf node views more reusable in other contexts; I don't think that really applies to this view. 

If it's more than that, and there's an efficiency factor to it, that would be interesting to understand more about.

Reply:
> One reason might just be to make the leaf node views more reusable in other contexts; I don't think that really applies to this view.

Yes, I think that's for reusability. If you have a button it's more reusable if it's configurable with the style directly rather than depending on an environment object.